### DISCUSSION TITLE
Delete just one image from a record with multiple images

### DISCUSSION THREAD

**Discussion by sawyer1776** [pocketbase/pocketbase#2210] [Q&A] [answered]
Right now I have multiple images  in an images field of a record, I am displaying them and I want the user to be able to delete just one of these images.  All I see in the docs is how to delete the record, or set the imgs field to null to delete all imgs not just one img.  Ideally I was thinking I get the index of the image in the array of images and then remove that, but I am not sure how to do that with pocketbase.  Any ideas?

**Comment by ganigeorgiev** [comment, marked as answer]
You can check https://pocketbase.io/docs/files-handling/#deleting-files.

You can use the `-` field data modifier and provide the filename(s) you want to delete. For example:

```js
await pb.collection('example').update('RECORD_ID', {
    'myFileField-': ['file1.pdf', 'file2.txt'],
    // or submit a new array with only the files you want to keep
    // 'myFileField': ["file3.txt", "file4.txt"]
});
```

_Note that for backward compatibility with earlier PocketBase versions we also support deleting by index suffix, eg. `'myFileField.0': null`, but it is recommended to use the above filename option as it is more clear which exact file(s) you are deleting._

**Reply by sawyer1776** [reply]
thanks so much! I will do this.
